main/test/ca/uwaterloo/flix/library/TestList.flix

```diff
@@ -0,0 +1,1836 @@
+
+@test
+def isSuffixOf07: Bool = assert!(List.isSuffixOf(Nil, 1 :: 2 :: Nil))
+
+@test
+def isSuffixOf08: Bool = assertNot!(List.isSuffixOf(1 :: Nil, 1 :: 2 :: Nil))
+
+@test
+def isSuffixOf09: Bool = assert!(List.isSuffixOf(2 :: Nil, 1 :: 2 :: Nil))
+
+@test
+def isSuffixOf10: Bool = assert!(List.isSuffixOf(1 :: 2 :: Nil, 1 :: 2 :: Nil))
+
+@test
+def isSuffixOf11: Bool = assertNot!(List.isSuffixOf(1 :: 3 :: Nil, 1 :: 2 :: Nil))
+
+@test
+def isSuffixOf12: Bool = assertNot!(List.isSuffixOf(1 :: 2 :: 3 :: Nil, 1 :: 2 :: Nil))
+
+@test
+def isSuffixOf13: Bool = assert!(List.isSuffixOf(1 :: 2 :: 3 :: Nil, 89 :: 11 :: 1 :: 2 :: 3 :: Nil))
+
+/////////////////////////////////////////////////////////////////////////////
+// fold                                                                    //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def fold01: Bool = List.fold((i, e) -> (i - e)*(e % 2 + 1), 100, Nil) `assertEq!` 100
+
+@test
+def fold02: Bool = List.fold((i, e) -> (i - e)*(e % 2 + 1), 100, 1 :: Nil) `assertEq!` 198
+
+@test
+def fold03: Bool = List.fold((i, e) -> (i - e)*(e % 2 + 1), 100, 1 :: 2 :: Nil) `assertEq!` 196
+
+@test
+def fold04: Bool = List.fold((i, e) -> (i - e)*(e % 2 + 1), 100, 1 :: 2 :: 3 :: Nil) `assertEq!` 386
+
+/////////////////////////////////////////////////////////////////////////////
+// foldLeft                                                                //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldLeft01: Bool = List.foldLeft((i, e) -> (i - e)*(e % 2 + 1), 100, Nil) `assertEq!` 100
+
+@test
+def foldLeft02: Bool = List.foldLeft((i, e) -> (i - e)*(e % 2 + 1), 100, 1 :: Nil) `assertEq!` 198
+
+@test
+def foldLeft03: Bool = List.foldLeft((i, e) -> (i - e)*(e % 2 + 1), 100, 1 :: 2 :: Nil) `assertEq!` 196
+
+@test
+def foldLeft04: Bool = List.foldLeft((i, e) -> (i - e)*(e % 2 + 1), 100, 1 :: 2 :: 3 :: Nil) `assertEq!` 386
+
+/////////////////////////////////////////////////////////////////////////////
+// foldRight                                                               //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldRight01: Bool = List.foldRight((e, i) -> (i - e)*(e % 2 + 1), 100, Nil) `assertEq!` 100
+
+@test
+def foldRight02: Bool = List.foldRight((e, i) -> (i - e)*(e % 2 + 1), 100, 1 :: Nil) `assertEq!` 198
+
+@test
+def foldRight03: Bool = List.foldRight((e, i) -> (i - e)*(e % 2 + 1), 100, 1 :: 2 :: Nil) `assertEq!` 194
+
+@test
+def foldRight04: Bool = List.foldRight((e, i) -> (i - e)*(e % 2 + 1), 100, 1 :: 2 :: 3 :: Nil) `assertEq!` 382
+
+/////////////////////////////////////////////////////////////////////////////
+// reduceOpt                                                               //
+/////////////////////////////////////////////////////////////////////////////
+//@test
+//def reduceOpt01: Bool = List.reduceOpt((a, b) -> a-b, Nil) `assertEq!` None
```

Comment:
Yes, this is the bug I e-mailed about in #364. The fix is to add a type annotation somewhere. I will fix it.